src/request_blocks.cpp

```diff
@@ -112,10 +112,12 @@ namespace libtorrent
 
 		if (prefer_contiguous_blocks == 0 && !time_critical_mode)
 		{
-			int blocks_per_piece = t.torrent_file().piece_length() / t.block_size();
-			prefer_contiguous_blocks = c.statistics().download_payload_rate()
-				* t.settings().get_int(settings_pack::whole_pieces_threshold)
-				> t.torrent_file().piece_length() ? blocks_per_piece : 0;
+			int const blocks_per_piece = t.torrent_file().piece_length() / t.block_size();
+			prefer_contiguous_blocks
+				= (c.statistics().download_payload_rate()
+				> t.torrent_file().piece_length()
+				/ std::max(1, t.settings().get_int(settings_pack::whole_pieces_threshold)))
```

Comment:
instead of using ``max`` here, there should be a check for ``whole_pieces_threshold`` to be 0, and set ``prefer_contiguous_blocks`` to false in that case.

Reply:
I will try it today and fix potential issue. Thanks!

Reply:
I will try it today, and fix potential issue if possible. Thanks